### PR TITLE
[RISCV][GISEL] Legalize G_JUMP_TABLE and G_BRJT

### PR DESCRIPTION
**Pull request by michaelmaitland**

### CODE REVIEW

File: llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp [hunk 1]
```diff
@@ -252,6 +252,12 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST) {
   getActionDefinitionsBuilder({G_FCEIL, G_FFLOOR})
       .libcallFor({s32, s64});
 
+  getActionDefinitionsBuilder(G_JUMP_TABLE).legalFor({p0});
+
+  getActionDefinitionsBuilder(G_BRJT).legalIf([=](const LegalityQuery &Query) {
+    return typeIs(0, p0) && typeInSet(1, {s32, sXLen});
+  });
+
```
Comment (tschuett) [inline llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp:260]:
```.
legalFor({{p0, s32}, {p0, sXLen}});
```

Comment (topperc) [reply llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp:260]: As written this statment is always, you have to pass Query to typeIs and typeInSet. Other wise it's just checking if the address of the typeIs and typeInSet functions are non-null. Which of course they are.

Comment (topperc) [reply llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp:260]: Does the s32 case ever occur? IRTranslator::emitJumpTableHeader seems to use the integer size of the pointer to create the type for that operand.

**Comment by michaelmaitland** [comment]
Closing in favor of #71970

**Pull request closed without merging**